peps/pep-0780.rst

```diff
@@ -0,0 +1,447 @@
+
+Backwards Compatibility
+=======================
+
+This is a pure extension to the existing environment markers and does not
+affect existing environment markers or dependency specifications, hence there
+are no direct backwards compatibility concerns.
+
+However, the introduction of the feature has implications for a
+number of ecosystem tools, especially those which attempt to support
+examination of data in ``pyproject.toml`` and ``requirements.txt``.
+
+Audit and Update Tools
+----------------------
+
+A wide range of tools understand Python dependency data as expressed in
+``requirements.txt`` files. (e.g., Dependabot, Tidelift, etc)
+
+Such tools inspect dependency data and, in some cases, offer tool-assisted or
+fully automated updates.
+It is our expectation that no such tools would support the new environment
+markers at first, and broad ecosystem support could take many months or even
+some number of years to arrive.
+
+As a result, users of the new environment markers would experience a
+degradation in their workflows and tool support at the time that they start
+using them. This is true of any new standard for where and how dependency data
+are encoded.
+
+Security Implications
+=====================
+
+This PEP introduces new syntax for specifying dependency information in
+projects. However, it does not introduce newly specified mechanisms for
+handling or resolving dependencies.
+
+It therefore does not carry security concerns other than those inherent in any
+tools which may already be used to install dependencies---i.e. malicious
+dependencies may be specified here, just as they may be specified in
+``requirements.txt`` files.
+
+How to Teach This
+=================
+
+The use of environment markers is well established and communicated chiefly
+in :ref:`packaging:dependency-specifiers`.
+The new environment marker can be introduced in the same document.
+Additionally, both for package authors and users, free-threading specific
+guidance can be provided at the `Python free-threading guide`_.
+The new ``sys.abi_features`` attribute will be documented in the Python
+standard library documentation.
+
+
+Reference Implementation
+========================
+
+The reference implementation for the ``sys.abi_features`` attribute can be
+found in `Add abi_features to sys
+<https://github.com/zklaus/cpython/pull/1>`__.
+
+The reference implementation for the environment markers is available in a fork
+of the ``packaging`` library at `Environment markers for ABI features
+<https://github.com/zklaus/packaging/pull/1>`__.
+
+`A demonstration package <https://github.com/zklaus/env-marker-example>`__ is
+also available.
+
+Since ``pip`` uses a vendored copy of ``packaging`` internally, we also provide
+`a patched version of pip
+<https://github.com/zklaus/pip/tree/env-marker-free-threading>`__, which is
+based on pypa/pip:main with the vendored ``packaging`` replaced by the
+reference implementation linked above.
```

Comment:
Editorial:
```suggestion
Since ``pip`` uses a vendored copy of ``packaging`` internally, we also provide
`a patched version of pip`__, which replaces the vendored ``packaging`` with the
reference implementation linked above.

__ https://github.com/zklaus/pip/tree/env-marker-free-threading
```